pandas/core/util/hashing.py

```diff
@@ -264,7 +287,7 @@ def hash_array(vals, encoding='utf8', hash_key=None, categorize=True):
 
         try:
             vals = hashing.hash_object_array(vals, hash_key, encoding)
-        except TypeError:
+        except (TypeError, ValueError):
```

Comment:
yes, that is related to how I coerce to an array above. When using `np.array([val])`, I don't get object arrays. But inferring the dtype with `infer_dtype_from_array` gives some overhead (but probably not too much)